lib/miq_environment.rb

```diff
@@ -94,6 +94,7 @@ def self.supports_command?(cmd)
 
       begin
         # If 'which apachectl' returns non-zero, it wasn't found
+        require "awesome_spawn"
```

Comment:
Does it maybe make sense to handle `LoadError` down in the `rescue`, so we fail hard instead of caching an invalid error like this?

* * *

Alternatively... we could just write this in Ruby:

```ruby
def which(cmd)
  ENV['PATH'].split(File::PATH_SEPARATOR).each do |path|
    cmd_path = File.join(path, cmd)
    
    return true if File.executable?(cmd_path) && !File.directory?(cmd_path)
  end
  
  false
end
```

- ref: https://github.com/hashicorp/vagrant/blob/main/lib/vagrant/util/which.rb
- ref ref: https://stackoverflow.com/a/5471032/3574689

Reply:
Still can't understand why this isn't baked into ruby.  Maybe we should add it to more_core_extensions